### PR TITLE
CONSOLE-3705: Phase 1 of using OpenShift Dynamic Plugin SDK

### PR DESCRIPTION
**Pull request by vojtechszocs**
This PR updates Console dynamic plugin **build-time** infrastructure to use [webpack](https://github.com/openshift/dynamic-plugin-sdk/tree/main/packages/lib-webpack) code from [OpenShift Dynamic Plugin SDK](https://github.com/openshift/dynamic-plugin-sdk).

Console **runtime** plugin infrastructure is updated to support both current (legacy) and new (standard) plugin manifest formats.

Console plugins built with updated `ConsoleRemotePlugin` will use the [standard `PluginManifest` format](https://github.com/openshift/dynamic-plugin-sdk/blob/main/packages/lib-core/src/types/plugin.ts).

There are two structural differences between Console legacy vs. standard plugin manifest format:

**1. Console specific properties**
- legacy: top-level properties in the manifest
```json
"displayName": "Console Demo Plugin",
"description": "Plasma reactors online. Initiating hyper drive.",
"disableStaticPlugins": ["@console/demo-plugin"]
```
- standard: scoped to `customProperties` object under the `console` namespace
```json
"customProperties": {
  "console": {
    "displayName": "Console Demo Plugin",
    "description": "Plasma reactors online. Initiating hyper drive.",
    "disableStaticPlugins": ["@console/demo-plugin"]
  }
}
```

**2. Additional build metadata**
- legacy: no properties, Console code uses hard-coded conventions
- standard: new top-level properties in the manifest
```json
"registrationMethod": "callback",
"baseURL": "/api/plugins/console-demo-plugin/",
"loadScripts": ["plugin-entry.js"],
"buildHash": "8740a41586612419c90da43d82c69fb8"
```

Updated `ConsoleRemotePlugin` should be backwards compatible. Its options can be omitted entirely:
```ts
new ConsoleRemotePlugin(),
```
or specified explicitly as needed:
```ts
new ConsoleRemotePlugin({
  // [added] pluginMetadata - default: parsed from `consolePlugin` object in `package.json` file
  // [added] extensions - default: parsed from `console-extensions.json` file
  // validateExtensionSchema - default: true
  // validateExtensionIntegrity - default: true
  // validateSharedModules - default: true
  // [deleted] validatePackageSchema - plugin metadata validation now done via Yup
}),
```
Env. variable `CONSOLE_PLUGIN_SKIP_EXT_VALIDATOR` is no longer supported by `ConsoleRemotePlugin` since we already have an explicit option for this purpose (`validateExtensionIntegrity`).

Other notable changes:
- `@openshift-console/dynamic-plugin-sdk-webpack` now depends on `@openshift/dynamic-plugin-sdk-webpack`
- removed support for **very** old Console plugins (where plugin SDK packages use webpack `5.0.0-beta.16`)

---

cc @spadgett @jhadvig

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "integration-tests"
   ],
   "scripts": {
-    "postinstall": "yarn generate",
+    "postinstall": "# TODO yarn generate",
```
Comment (vojtechszocs) [inline frontend/package.json:14]: This is a temporary change :smiley:

File: frontend/packages/console-dynamic-plugin-sdk/README.md [hunk 2]
```diff
@@ -100,8 +100,6 @@ a [semver pre-release](https://semver.org/#spec-item-9) identifier, adapt your s
 to include the relevant pre-release prefix, e.g. use `~4.11.0-0.ci` when targeting pre-release versions
 like `4.11.0-0.ci-1234`.
 
-See `ConsolePluginMetadata` type for details on the `consolePlugin` object and its schema.
```
Comment (vojtechszocs) [inline frontend/packages/console-dynamic-plugin-sdk/README.md:103]: This README needs update, I will do it once the local testing is done.

**Comment by vojtechszocs** [comment]
Tested locally on `dynamic-demo-plugin` and everything seems to work as expected for both
- demo plugin built with current Console plugin SDK
- demo plugin built with updated Console plugin SDK (with changes in this PR)

**Comment by vojtechszocs** [comment]
Ready for review, will test locally in the meantime.

**Comment by vojtechszocs** [comment]
This PR improves the fix for [OCPBUGS-3495](https://issues.redhat.com//browse/OCPBUGS-3495) by making sure the entry script filename is `plugin-entry.[contenthash].min.js` for production builds of Console dynamic plugins.

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Comment by TheRealJon** [comment]
/lgtm cancel

Looks like the dynamic demo plugin e2e tests are failing. Might be a legit regression.

**Comment by TheRealJon** [comment]
/retest

cluster install failed

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Comment by jhadvig** [comment]
```
level=info msg=Waiting up to 40m0s (until 9:58PM UTC) for the cluster at https://api.ci-op-mxqfn55k-75d12.xxxxxxxxxxxxxxxxxxxxxx:6443/ to initialize...
level=error msg=Cluster operator authentication Degraded is True with OAuthServerRouteEndpointAccessibleController_SyncError: OAuthServerRouteEndpointAccessibleControllerDegraded: Get "https://oauth-openshift.apps.ci-op-mxqfn55k-75d12.xxxxxxxxxxxxxxxxxxxxxx/healthz": dial tcp: lookup oauth-openshift.apps.ci-op-mxqfn55k-75d12.XXXXXXXXXXXXXXXXXXXXXX on 172.30.0.10:53: no such host (this is likely result of malfunctioning DNS server)
level=error msg=Cluster operator authentication Available is False with OAuthServerRouteEndpointAccessibleController_EndpointUnavailable: OAuthServerRouteEndpointAccessibleControllerAvailable: Get "https://oauth-openshift.apps.ci-op-mxqfn55k-75d12.XXXXXXXXXXXXXXXXXXXXXX/healthz": dial tcp: lookup oauth-openshift.apps.ci-op-mxqfn55k-75d12.XXXXXXXXXXXXXXXXXXXXXX on 172.30.0.10:53: no such host (this is likely result of malfunctioning DNS server)
level=info msg=Cluster operator baremetal Disabled is False with : 
level=info msg=Cluster operator cloud-controller-manager TrustedCABundleControllerControllerAvailable is True with AsExpected: Trusted CA Bundle Controller works as expected
level=info msg=Cluster operator cloud-controller-manager TrustedCABundleControllerControllerDegraded is False with AsExpected: Trusted CA Bundle Controller works as expected
level=info msg=Cluster operator cloud-controller-manager CloudConfigControllerAvailable is True with AsExpected: Cloud Config Controller works as expected
level=info msg=Cluster operator cloud-controller-manager CloudConfigControllerDegraded is False with AsExpected: Cloud Config Controller works as expected
level=info msg=Cluster operator cloud-controller-manager CloudControllerOwner is True with AsExpected: Cluster Cloud Controller Manager Operator owns cloud controllers at 4.15.0-0.ci.test-2023-11-22-[205](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_console/13188/pull-ci-openshift-console-master-e2e-gcp-console/1727429461328007168#1:build-log.txt%3A205)139-ci-op-mxqfn55k-latest
level=info msg=Cluster operator console Progressing is True with SyncLoopRefresh_InProgress: SyncLoopRefreshProgressing: Working toward version 4.15.0-0.ci.test-2023-11-22-205139-ci-op-mxqfn55k-latest, 0 replicas available
level=error msg=Cluster operator console Available is False with Deployment_InsufficientReplicas::RouteHealth_FailedGet: DeploymentAvailable: 0 replicas available for console deploymen
```

/retest

**Comment by yanpzhan** [comment]
@vojtechszocs I checked on cluster launched against the pr, and deployed console-crontab-plugin and console-demo-plugin successfully. If there are other checkpoints need to be covered?

**Comment by vojtechszocs** [comment]
> @vojtechszocs I checked on cluster launched against the pr, and deployed console-crontab-plugin and console-demo-plugin successfully. If there are other checkpoints need to be covered?

Hi @yanpzhan - I think the above basic checks should be enough :smiley:

**Comment by yanpzhan** [comment]
Thanks all.
/label qe-approved

**Pull request merged**